packages/client/components/ActivityLibrary/ScheduleMeetingButton.tsx

```diff
@@ -0,0 +1,106 @@
+import React, {useEffect, useState} from 'react'
+import graphql from 'babel-plugin-relay/macro'
+import {useDialogState} from '../../ui/Dialog/useDialogState'
+import SecondaryButton from '../SecondaryButton'
+import GcalModal from '../../modules/userDashboard/components/GcalModal/GcalModal'
+import {CreateGcalEventInput} from '../../__generated__/StartRetrospectiveMutation.graphql'
+import GcalClientManager from '../../utils/GcalClientManager'
+import useAtmosphere from '../../hooks/useAtmosphere'
+import {useFragment} from 'react-relay'
+import {ScheduleMeetingButton_team$key} from '~/__generated__/ScheduleMeetingButton_team.graphql'
+import {ScheduleMeetingButton_viewer$key} from '~/__generated__/ScheduleMeetingButton_viewer.graphql'
+import {MenuMutationProps} from '../../hooks/useMutationProps'
+
+type Props = {
+  mutationProps: MenuMutationProps
+  handleStartActivity: (gcalInput?: CreateGcalEventInput) => void
+  teamRef: ScheduleMeetingButton_team$key
+  viewerRef: ScheduleMeetingButton_viewer$key
+}
+
+const ScheduleMeetingButton = (props: Props) => {
+  const {mutationProps, handleStartActivity, teamRef, viewerRef} = props
+  const {
+    isOpen: isScheduleDialogOpen,
+    open: openScheduleDialog,
+    close: closeScheduleDialog
+  } = useDialogState()
+  const atmosphere = useAtmosphere()
+  const [hasStartedGcalAuthTeamId, setHasStartedGcalAuthTeamId] = useState<null | string>(null)
+  const {submitting} = mutationProps
+
+  const viewer = useFragment(
+    graphql`
+      fragment ScheduleMeetingButton_viewer on User {
+        featureFlags {
+          gcal
+        }
+      }
+    `,
+    viewerRef
+  )
+  const hasGcalFlag = viewer.featureFlags.gcal
+
+  const team = useFragment(
+    graphql`
+      fragment ScheduleMeetingButton_team on Team {
+        id
+        teamMembers {
```

Comment:
Ideally, would be good to not fetch all teamMembers and their integrations, as we only need one

Reply:
-1 you can query `teamMember(teamId: ID!)` on `viewer` instead, can't you? I think introducing a `teamMember` field which returns the viewer's teamMember would also be an option.